linkerd/docs/protocol-http.md

```diff
@@ -165,6 +165,47 @@ Key | Default Value | Description
 dstPrefix | `http` | The `dstPrefix` as set in the routers block.
 urlPath | N/A | A path from the URL whose number of segments is set in the identifier block.
 
+<a name="header-identifier"></a>
+### Header Identifier
+
+kind: `io.l5d.header`
+
+With this identifier, HTTP requests are turned into names based only on the
+value of an HTTP header.  If the header value is a valid path, that path is
+used.  Otherwise, the header value is converted to a path with one path segment.
+
+#### Namer Configuration:
+
+> With this configuration, the value of the `my-header` HTTP header will be used
+as the logical name.
+
+```yaml
+routers:
+- protocol: http
+  identifier:
+    kind: io.l5d.header
+    header: my-header
+  servers:
+  - port: 5000
+```
+
+Key | Default Value | Description
+--- | ------------- | -----------
+header | `l5d-dst-concrete` | The name of the HTTP header to use
```

Comment:
I'm find this naming to be unclear. Elsewhere, we describe pre-delegation names (i.e. Dst.Path) as _logical_ and post-delegation names (i.e. Dst.Id) as _concrete_.  In this context, the value isn't really a concrete name.  We should try to reconcile this.
- `l5d-dst-name`?
- `l5d-dst-path?
- ?